### PR TITLE
fcitx5: 5.1.1 -> 5.1.5

### PR DESCRIPTION
**Pull request by Vonfry**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Vonfry** [comment]
Result of `nixpkgs-review pr 265589` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>fcitx5</li>
    <li>fcitx5-anthy</li>
    <li>fcitx5-chewing</li>
    <li>fcitx5-chinese-addons</li>
    <li>fcitx5-configtool</li>
    <li>fcitx5-gtk</li>
    <li>fcitx5-hangul</li>
    <li>fcitx5-lua</li>
    <li>fcitx5-m17n</li>
    <li>fcitx5-mozc</li>
    <li>fcitx5-openbangla-keyboard</li>
    <li>fcitx5-rime</li>
    <li>fcitx5-skk</li>
    <li>fcitx5-skk-qt</li>
    <li>fcitx5-table-extra</li>
    <li>fcitx5-table-other</li>
    <li>fcitx5-unikey</li>
    <li>fcitx5-with-addons</li>
    <li>libime</li>
    <li>libsForQt5.fcitx5-qt</li>
    <li>mlterm</li>
    <li>mlterm-wayland</li>
  </ul>
</details>

File: pkgs/tools/inputmethods/fcitx5/fcitx5-qt.nix [hunk 1]
```diff
@@ -9,17 +9,22 @@
...
 , qt6
+, wayland
 }:
 
-mkDerivation rec {
+let
+  qt6env = qt6.env "qt-fcitx5-env-${qt6.qtbase.version}" [
+    qt6.qtbase qt6.qtwayland
+  ];
+in mkDerivation rec {
```
Comment (NickCao) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-qt.nix:19]:
```suggestion
mkDerivation rec {
```

File: pkgs/tools/inputmethods/fcitx5/fcitx5-qt.nix [hunk 2]
```diff
@@ -31,7 +36,7 @@ mkDerivation rec {
 
   cmakeFlags = [
     # adding qt6 to buildInputs would result in error: detected mismatched Qt dependencies
-    "-DCMAKE_PREFIX_PATH=${qt6.qtbase}"
+    "-DCMAKE_PREFIX_PATH=${qt6env}"
```
Comment (NickCao) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-qt.nix:39]:
```suggestion
    "-DCMAKE_PREFIX_PATH=${qt6.qtbase};${qt6.qtwayland}"
```

**Review by NickCao — COMMENTED** [review]
This seem to reduce the closure size by a little bit.

**Review by NickCao — APPROVED** [review]
Result of `nixpkgs-review pr 265589` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>fcitx5</li>
    <li>fcitx5-anthy</li>
    <li>fcitx5-chewing</li>
    <li>fcitx5-chinese-addons</li>
    <li>fcitx5-configtool</li>
    <li>fcitx5-gtk</li>
    <li>fcitx5-hangul</li>
    <li>fcitx5-lua</li>
    <li>fcitx5-m17n</li>
    <li>fcitx5-mozc</li>
    <li>fcitx5-openbangla-keyboard</li>
    <li>fcitx5-rime</li>
    <li>fcitx5-skk</li>
    <li>fcitx5-skk-qt</li>
    <li>fcitx5-table-extra</li>
    <li>fcitx5-table-other</li>
    <li>fcitx5-unikey</li>
    <li>fcitx5-with-addons</li>
    <li>libime</li>
    <li>libsForQt5.fcitx5-qt (plasma5Packages.fcitx5-qt)</li>
    <li>mlterm</li>
    <li>mlterm-wayland</li>
  </ul>
</details>

Tested `chineseaddon` as well as setting `GTK_IM_MODULE=wayland`.

**Comment by Vonfry** [comment]
rebose and update fcitx5 to 1.5.5

**Comment by Vonfry** [comment]
rebese onto master and update fcitx5 to 1.5.5

**Comment by oxalica** [comment]
Just a note: I tested 5.1.4 and 5.1.4 several days, and found that telegram (and probably other QT6 programs) suffers a lot of freezing when typing quickly with fcitx5 under swaywm. The author of fcitx5 blames QT for the issue [QTBUG-118650](https://bugreports.qt.io/browse/QTBUG-118650).
My own experience is that the bug appears on 5.1.4 or 5.1.5 much more frequently than 5.1.1 such that fcitx5 is almost unusable under QT6 programs.

**Comment by oxalica** [comment]
> The author of fcitx5 blames QT for the issue [QTBUG-118650](https://bugreports.qt.io/browse/QTBUG-118650).

After reverting [the commit](https://code.qt.io/cgit/qt/qtwayland.git/patch/?id=8ed9c8279fd568ce4222e196847b90ac4e362ee8) mentioned in that QTBUG, telegram does not freeze anymore for me. Should we revert it for `qt6.qtwayland`, or wait for a more fix elegant fix? That commit itself is a fix to another issue, according to its commit message.

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 265589` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>fcitx5</li>
    <li>fcitx5-anthy</li>
    <li>fcitx5-chewing</li>
    <li>fcitx5-chinese-addons</li>
    <li>fcitx5-configtool</li>
    <li>fcitx5-gtk</li>
    <li>fcitx5-hangul</li>
    <li>fcitx5-lua</li>
    <li>fcitx5-m17n</li>
    <li>fcitx5-mozc</li>
    <li>fcitx5-openbangla-keyboard</li>
    <li>fcitx5-rime</li>
    <li>fcitx5-skk</li>
    <li>fcitx5-skk-qt</li>
    <li>fcitx5-table-extra</li>
    <li>fcitx5-table-other</li>
    <li>fcitx5-unikey</li>
    <li>fcitx5-with-addons</li>
    <li>libime</li>
    <li>libsForQt5.fcitx5-qt (plasma5Packages.fcitx5-qt)</li>
    <li>mlterm</li>
    <li>mlterm-wayland</li>
  </ul>
</details>

**Pull request merged**